tests/cxx/declare_symbol.h

```diff
@@ -0,0 +1,13 @@
+//===--- declare_symbol.h - test input file for iwyu ----------------------===//
```

Comment:
OK, so I tested. It does work (see `internal/internal_files.cc`), but I wouldn't group things with directories unless it makes a difference for the IWYU analysis.